bindings/pyroot/src/Converters.cxx

```diff
@@ -1013,24 +1013,54 @@ Bool_t PyROOT::TValueCppObjectConverter::SetArg(
 {
-   if ( ! ObjectProxy_Check( pyobject ) )
-      return kFALSE;
+   if ( ObjectProxy_Check( pyobject ) ) {   // It is PyROOT object
+      ObjectProxy* pyobj = (ObjectProxy*)pyobject;
+      if ( pyobj->ObjectIsA() && Cppyy::IsSubtype( pyobj->ObjectIsA(), fClass ) ) {
+      // calculate offset between formal and actual arguments
+            para.fValue.fVoidp = pyobj->GetObject();
+            if ( pyobj->ObjectIsA() != fClass ) {
+            para.fValue.fLong += Cppyy::GetBaseOffset(
+                  pyobj->ObjectIsA(), fClass, para.fValue.fVoidp, 1 /* up-cast */ );
+            }
 
-   ObjectProxy* pyobj = (ObjectProxy*)pyobject;
-   if ( pyobj->ObjectIsA() && Cppyy::IsSubtype( pyobj->ObjectIsA(), fClass ) ) {
-   // calculate offset between formal and actual arguments
-      para.fValue.fVoidp = pyobj->GetObject();
-      if ( pyobj->ObjectIsA() != fClass ) {
-         para.fValue.fLong += Cppyy::GetBaseOffset(
-            pyobj->ObjectIsA(), fClass, para.fValue.fVoidp, 1 /* up-cast */ );
+            para.fTypeCode = 'V';
+            return kTRUE;
+      } else if ( ! TClass::GetClass( Cppyy::GetFinalName( fClass ).c_str() )->GetClassInfo() ) {
+      // assume "user knows best" to allow anonymous reference passing
+            para.fValue.fVoidp = pyobj->GetObject();
+            para.fTypeCode = 'V';
+            return kTRUE;
+      }
+   }
+   else if ( PyTuple_Check( pyobject ) ){  // It is a Python tuple (equivalent to a C++ initializer list)
+
+      // instantiate an object proxy of this class
+      if( ! fObjProxy ) {
+      // retrieve python class
+         PyObject* pyclass = CreateScopeProxy( fClass );
+         if ( ! pyclass ) return kFALSE;                  // error has been set in CreateScopeProxy
+         fObjProxy = (ObjectProxy*)((PyTypeObject*)pyclass)->tp_new( (PyTypeObject*)pyclass, pyobject, NULL );
+         Py_DECREF( pyclass );
       }
 
+      if( fObjProxy->GetObject() ) {
+         Cppyy::CallDestructor( fObjProxy->ObjectIsA(), fObjProxy->GetObject() );
+         Cppyy::Deallocate( fObjProxy->ObjectIsA(), fObjProxy->GetObject() );
+         fObjProxy->Set(nullptr);
+      }
+
+      // get the constructor (i.e. __init__)
+      PyObject* constructor = PyObject_GetAttr( (PyObject*)fObjProxy, PyStrings::gInit );
+      if( ! constructor ) return kFALSE; 
+      // call the constructor with the arguments in the given tuple
+      PyObject* obj = PyObject_CallObject( constructor, pyobject );
+      Py_DECREF( constructor );
+      if ( ! obj ) return kFALSE;
+      Py_DECREF( obj );
+
+      para.fValue.fVoidp = fObjProxy->GetObject();
       para.fTypeCode = 'V';
-      return kTRUE;
-   } else if ( ! TClass::GetClass( Cppyy::GetFinalName( fClass ).c_str() )->GetClassInfo() ) {
-   // assume "user knows best" to allow anonymous reference passing
-      para.fValue.fVoidp = pyobj->GetObject();
-      para.fTypeCode = 'V';
+      // TODO we need to remove the temporary object constructed from the tuple
```

Comment:
This is done in lines 1047 - 1049

Reply:
Yes. The comment can be deleted.

Reply:
Yes sure! I remembered that we needed to have this put aside as we did for strings :)
Thanks.